src/Controller/MakefileTraceToJobTree.ts

```diff
@@ -0,0 +1,210 @@
+import { strict as assert } from 'assert'
+import { Job } from '../Job/Job'
+import { NormalJob } from '../Job/NormalJob'
+
+/**
+ * A Job and whether or not it has a dependent.
+ *
+ * Used during DAG construction.
+ */
+interface JobAndDependency {
+	job: Job
+	hasDependent: boolean
+}
+
+/**
+ * A list of target-lines and their associated command-lines.
+ *
+ * Used during trace parsing.
+ */
+interface TargetLineAndCommands {
+	targetLine: string
+	commands: string[]
+}
+
+// Regexes used when interpreting target-lines.
+const targetLineRegexWithPrereqs = /^.+: update target '(.+)' due to: (.+)$/
```

Comment:
These regexes seem dangerous. Can we replace this with something like: 

```suggestion
const targetLineRegexWithPrereqs = /^.+: update target '([A-Za-z0-9_/.]+)' due to: (([A-Za-z0-9_/.]+)$/
```

Reply:
I updated the regex.